core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolutionEditorDirective.js

```diff
@@ -205,8 +189,13 @@ oppia.directive('stateSolutionEditor', [
                 stateContentIdsToAudioTranslationsService.displayed
                   .addContentId(explanationContentId);
                 stateContentIdsToAudioTranslationsService.saveDisplayedValue();
+                $scope.onSaveContentIdsToAudioTranslations(
+                  stateContentIdsToAudioTranslationsService.displayed
+                );
               }
               stateSolutionService.saveDisplayedValue();
+              $scope.onSaveSolution(stateSolutionService.displayed);
+              $rootScope.$broadcast('validateSolution', false);
```

Comment:
No, try not to do this, it's effectively a goto statement

Reply:
Ok, for this, I can have another function passed in, but what about the one in ResponsesService? Also, I couldn't keep it there, since it used exploration specific services.

Reply:
Done.